### PR TITLE
Adding a call back in MultiApp

### LINKED ISSUE #5987: Require a call back in MultiApp
I need a call back for adding an action before ```app->runInputFile()``` is executed to add my own 
action.

### PR DESCRIPTION
**Pull request by snschune**
closes #5987

### CODE REVIEW

File: framework/src/multiapps/MultiApp.C [hunk 1]
```diff
@@ -502,6 +502,7 @@ MultiApp::createApp(unsigned int i, Real start_time)
   // Update the MultiApp level for the app that was just created
   app->setMultiAppLevel(_app.multiAppLevel() + 1);
   app->setupOptions();
+  preRunInputFile();
```
Comment (friedmud) [inline framework/src/multiapps/MultiApp.C]: Who should be getting called here?  This looks out of place.  It seems like this should be calling something in the `App`...

Comment (YaqiWang) [inline framework/src/multiapps/MultiApp.C]: Can a MooseApp know it's wrapped by a MultiApp? If so, we could do some special setup in a wrapped MooseApp based on the master app.

Comment (snschune) [inline framework/src/multiapps/MultiApp.C]: @friedmud Let me just do a quick explanation what I want to do with this function and why it's here. I have a master mesh nested in the sub app mesh. The transfers are such that each sub app element that's in a master app element must get the same procedure ID from the partitioner. For ensuring that I have an EmbeddedMeshPartitioner. However, the EmbeddedMeshPartitioner requires the master mesh as input and is part of the sub app. Therefore, the multiapp is really the only place I could put it as I have access to both master and sub app stuff. I programmatically add an action in this call back that sets the correct partitioner and its parameters.

Comment (friedmud) [inline framework/src/multiapps/MultiApp.C]: @YaqiWang : no, it's part of the design that a sub app doesn't know that it's a sub app.  That is on purpose and is done for your own good ;-)

@snschune : ok.  This feels a little specific, but it's not a big deal.  I'm not against this.

**Pull request merged**